src/test/java/school/lemon/changerequest/java/junit/hw1/task1/api/ContactApiTest.java

```diff
@@ -0,0 +1,215 @@
+        assertTrue(result);
+        assertEquals(2, contactApi.filledContactsAmount());
+        Contact[] afterDelete = contactApi.getAll();
+
+        assertEquals(list[0], afterDelete[0]);
+        assertEquals(list[2], afterDelete[1]);
+
+        contactApi.delete((long) 1002);
+        assertEquals(1, contactApi.filledContactsAmount());
+        afterDelete = contactApi.getAll();
+
+        assertEquals(list[0], afterDelete[0]);
+
+        contactApi.delete((long) 1000);
+        assertEquals(0, contactApi.filledContactsAmount());
+    }
+
+    @Test
+    public void deleteNotExistedContactTest() {
+        boolean result = contactApi.delete((long) 0);
+        assertFalse(result);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void deleteNullTest() {
+        contactApi.delete(null);
+    }
+
+
+    @Test
+    public void updateTest() {
+        Contact beforeUpdate = contactApi.get((long) 1002);
+        beforeUpdate.setName("TestName");
+        beforeUpdate.setSurname("TestSurName");
+        boolean result = contactApi.update(beforeUpdate);
+        Contact afterUpdate = contactApi.get((long) 1002);
+        assertEquals(beforeUpdate, afterUpdate);
+        assertTrue(result);
+    }
+
+    @Test
+    public void updateNotExistedContactTest() {
+        boolean result = contactApi.update(testContact);
+        assertFalse(result);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void updateNullTest() throws Exception {
+        contactApi.update(null);
+    }
+
+    @Test
+    public void createTest() {
+        assertEquals(3, contactApi.filledContactsAmount());
+        try {
+            contactApi.find(testContact.getCallNumber());
+            fail();
+        } catch (ContactNotFoundException e) {
+        }
+
+        contactApi.create(testContact);
+        assertEquals(4, contactApi.filledContactsAmount());
+
+        Contact result = contactApi.find(testContact.getCallNumber());
+        assertEquals(testContact, result);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void createNullTest() {
+        contactApi.create(null);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void createFullArrayTest() {
+        fillWithContacts();
+        Contact contact = new Contact((long) 0, new CallNumber(randomNumeric(2), randomNumeric(3), randomNumeric(8)), "Test", "Test");
+        contactApi.create(contact);
+    }
+
+    @Test
+    public void getAllTest() {
+        Contact[] list = contactApi.getAll();
+        initContactsTest(list);
+
+        contactApi.create(testContact);
+        list = contactApi.getAll();
+        initContactsTest(list);
+        assertEquals(testContact, list[3]);
+    }
+
+    @Test
+    public void getAllEmptyListTest() {
+        deleteAllInitContacts();
+        Contact[] emptyList = contactApi.getAll();
+        assertEquals(0, emptyList.length);
+    }
+
+    private void initContactsTest(Contact[] list) {
```

Comment:
What if I'll add additional test case and pass you an array with only 2 elements?

Your test classes code should have the same quality as your code under tests.